code/obj/item/plants.dm

```diff
@@ -437,15 +437,30 @@
 	desc = "By any other name, would smell just as sweet. This one likes to be called "
 	icon_state = "rose"
 	var/thorned = 1
-	var/list/names = list("Emma", "Olivia", "Ava", "Isabella", "Sophia", "Charlotte", "Mia", "Amelia",
-	"Harper", "Evelyn", "Abigail", "Emily", "Elizabeth", "Mila", "Dakota", "Avery",
-	"Sofia", "Camila", "Aria", "Scarlett", "Liam", "Noah", "William", "James",
-	"Oliver", "Benjamin", "Elijah", "Lucas", "Mason", "Logan", "Alexander", "Ethan",
-	"Jacob", "Michael", "Daniel", "Henry", "Jackson", "Sebastian", "Aiden", "Matthew")
 
 	New()
 		..()
-		desc = desc + pick(names) + "."
+		var/list/possible_names = list()
+		for(var/mob/living/carbon/human/H in mobs)
+			if(isnpcmonkey(H))
+				continue
+			if(iswizard(H))
+				continue
+			if(isnukeop(H))
+				continue
+			if(!H.mind)
+				continue
+			possible_names += H
+		for(var/mob/living/silicon/ai/A in mobs)
+			possible_names += A
```

Comment:
Slight nitpick- `A` is usually for atoms. would rather call this `aimob` or something similar
Also- why AIs and not cyborgs? simple way would be iterating on all `/mob/living/silicon`, but that would also catch ghost drones (if you care about that).
Alternately just do this loop but for cyborgs as well (`/mob/living/silicon/robot`)

Reply:
oh also check for a mind again so you don't get a million Cyborg the Rose when the roboticists are going mental on the fabs

Reply:
Sure, changed to ``aimob`` instead of ``A``. Added a robot loop to include borgs but avoid ghostdrones and added the check for mind to avoid empty shells being chosen. I'd like to try implementing a robotic-rose mutation to handle silicons in the future but I'm happy with this for now